test/unit/sql/insert.test.js

```diff
@@ -109,10 +104,11 @@ describe(Support.getTestDialectTeaser('SQL'), () => {
         timestamps: true
       });
 
-      expectsql(sql.bulkInsertQuery(User.tableName, [{ user_name: 'testuser', pass_word: '12345' }], { updateOnDuplicate: ['username', 'password', 'updatedAt'] }, User.rawAttributes),
+      expectsql(sql.bulkInsertQuery(User.tableName, [{ user_name: 'testuser', pass_word: '12345' }], { updateOnDuplicate: ['user_name', 'pass_word', 'updated_at'] }, User.fieldRawAttributesMap),
```

Comment:
a breaking change?

What this test assumes seems incorrect to me, if records are mapped by field name so should `updateOnDuplicate` values. `bulkInsertQuery` should just use whatever data given to it to generate query as its a low level API, converting attributes to fields etc stuff should be handled at model level api.

I think it should be ok to merge this change, `records` are expected to be mapped by field name so should `updateOnDuplicate`